drivers/video/gc2145.c

```diff
@@ -1105,11 +1139,12 @@ static int gc2145_init(const struct device *dev)
 	gc2145_soft_reset(dev);
 	gc2145_write_all(dev, default_regs, ARRAY_SIZE(default_regs));
 
-	/* set default/init format SVGA RGB565 */
+	/* set default/init format QVGA RGB565 */
 	fmt.pixelformat = VIDEO_PIX_FMT_RGB565;
-	fmt.width = UXGA_HSIZE;
-	fmt.height = UXGA_VSIZE;
-	fmt.pitch = UXGA_HSIZE * 2;
+	fmt.width = fmts[QVGA_RESOLUTION].width_min;
+	fmt.height = fmts[QVGA_RESOLUTION].height_min;
+	fmt.pitch = fmts[QVGA_RESOLUTION].width_min * 2;
```

Comment:
It should not be a problem, the pitch should be provided by the caller, and then it gets copied to the driver data when setting the resolution.

Reply:
[here](https://github.com/zephyrproject-rtos/zephyr/pull/78422/files#diff-83d2e8ab89cc9ed3403bffe5472f4c642b331b0d887612c3bdc8696089e82b3aR1043) I suppose
```c
drv_data->fmt = *fmt;
```

But it looks ignored altogether by the driver: no custom pitch available it seems.

Reply:
Yes, this line would act like a memcpy from the input parameter to the driver data struct, in the sensor driver the pitch itself is not used in runtime, but in the default state it does, because most of video applications may fetch the default format via video_get_caps, and the pitch is used to calculate the amount of space in the video buffer.